Revit_Core_Engine/Convert/Revit/FromRevit/ViewFilter.cs

```diff
@@ -0,0 +1,486 @@
+
+            // Convert the REVIT FilterNumericEvaluator type into the BHOM NumberComparisonType Enum 
+            NumberComparisonType bhomNumericEvaluator=NumberComparisonTypeFromRevit(numericEvaluator.GetType().Name);
+
+            // 2. BUILD the BHOM FILTERRULE object
+
+            oM.Revit.FilterRules.FilterIntegerRule bhomFilterIntegerRule;
+            bhomFilterIntegerRule = new BH.oM.Revit.FilterRules.FilterIntegerRule();
+                            bhomFilterIntegerRule.ParameterName = paramName;
+                            bhomFilterIntegerRule.Value = paramValue;
+                            bhomFilterIntegerRule.ComparisonType = bhomNumericEvaluator;
+
+            return bhomFilterIntegerRule;
+        }
+
+
+        private static oM.Revit.FilterRules.FilterElementIdRule FilterRuleFromRevit(this ParameterFilterElement revitViewFilter,
+                    Autodesk.Revit.DB.FilterElementIdRule revitRule)
+        {
+            // 1. EXTRACT DATA from the REVIT FILTERRULE object
+
+            // Downcast Revit FilterRule obj to Revit FilterElementIdRule obj
+            Autodesk.Revit.DB.FilterElementIdRule revitFilterElemIdRule = (Autodesk.Revit.DB.FilterElementIdRule)revitRule;
+            // Extract name and value assigned to the parameter of the FilterElementIdRule obj
+            string paramName = GetParameterById(revitViewFilter.Document, revitRule.GetRuleParameter()).Definition.Name;
+            string paramValue = revitViewFilter.Document.GetElement(revitRule.RuleValue).Name.ToString();
+            // Get the RuleEvaluator of the FilterElementIdRule (Class defining the way the string value 
+            // assigned to the parameter is compared with the one assigned by the filter)
+            FilterNumericRuleEvaluator numericEvaluator = revitRule.GetEvaluator();
+
+            // Convert the REVIT FilterNumericEvaluator type into the BHOM NumberComparisonType Enum 
+            NumberComparisonType bhomNumericEvaluator = NumberComparisonTypeFromRevit(numericEvaluator.GetType().Name);
+
+            // 2. BUILD the BHOM FILTERRULE object
+
+            oM.Revit.FilterRules.FilterElementIdRule bhomFilterElemIdRule;
+            bhomFilterElemIdRule = new BH.oM.Revit.FilterRules.FilterElementIdRule();
+            bhomFilterElemIdRule.ParameterName = paramName;
+            bhomFilterElemIdRule.Value = paramValue;
+            bhomFilterElemIdRule.ComparisonType = bhomNumericEvaluator;
+
+            return bhomFilterElemIdRule;
+        }
+
+
+        private static oM.Revit.FilterRules.FilterCategoryRule FilterRuleFromRevit(this ParameterFilterElement revitViewFilter, Autodesk.Revit.DB.FilterCategoryRule revitRule)
+        {
+            // 1. EXTRACT DATA from the REVIT FILTERRULE object
+
+            // Downcast Revit FilterRule obj to Revit FilterCategoryRule obj
+            Autodesk.Revit.DB.FilterCategoryRule revitFilterCategoryRule = (Autodesk.Revit.DB.FilterCategoryRule)revitRule;
```

Comment:
No need to do this casting any more (refers to all private methods that do it in this file)

Reply:
Removed all the casts in the code.